site/v5/testing/testing.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Testing  
+date: 2022-05-02 11:30:00 
+permalink: /v5/testing/index.html
+eleventyNavigation:
+  version: v5
+  root: true
+  order: 80
+---
+<h1 class="title">Testing</h1>
+
+[[TOC]]
+
+# Introduction
+This sections exaplains the different levels of testing(unit and integration tests) that can(and should) be applied to a Mongock migration and the tools Mongock provides for it.
+
+# Unit testing
+This is probably the easiest and more basic level of testing, which takes a changeUnit in isolation and ensures it performs the right actions and calls to the dependend components. Although it's far from being enough, as the components are normally injected as mock/stub/spy, it's a good starting point to ensure that the correctness of the changeUnit.
+
+Mongock doesn't provide any speicific tool for this, but we illustrate how to do it  [in our example project](https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-quickstart)
+
+A unit test for a change unit looks like this:
+
+<p class="text-center">
+    <img src="/images/changeUnit-unit-test.png" alt="ChangeUnit unit test">
+</p>
+
+
+# Integration test
+At this point, once we are confident our changeUnits do what they should, we probably want to test our migration(set of changeUnits) within the application  context and, given the specific input scenarios, the result in the database is what we expect.
+
+This is a more complex level of testing as it requires to simulate the application context and implies the integration of the different components within the application. But it's probably the most important level of testing to ensure the correctness of the migration.
+
+To see an example, please see [our example project](https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-quickstart)
+
+
+## Integration test with Springboot Runner with Junit5
+Mongock provides some util classes to make this pretty easy. In summary you need to create your test class extending `MongockSpringbootJUnit5IntegrationTestBase`, which provides the following
+- **BeforeEach method(automatically called):** Resets mongock to allow re-utilization(not recommended in production) and build the runner
```

Comment:
"Not recommended in production" - Why is this? If there is a good reason, we should also put it as a red note as part of the main documentation of the method given it can be dangerous.

Reply:
A Mongock Runner cannot be re-used. However for testing purposed this is enabled to be able to run the same runner against multiple scenarios. That's why it says "utilisation is not recommended in production"